providers/transip/transipProvider.go

```diff
@@ -109,47 +108,138 @@ func (n *transipProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*mode
 
 	models.PostProcessRecords(curRecords)
 
+	if !diff2.EnableDiff2 {
+		corrections, err := n.getCorrectionsUsingOldDiff(dc, curRecords)
+		return corrections, err
+	}
+
+	corrections, err := n.getCorrectionsUsingDiff2(dc, curRecords)
+	return corrections, err
+}
+
+func (n *transipProvider) getCorrectionsUsingDiff2(dc *models.DomainConfig, records models.Records) ([]*models.Correction, error) {
 	var corrections []*models.Correction
-	if !diff2.EnableDiff2 || true { // Remove "|| true" when diff2 version arrives
+	instructions, err := diff2.ByRecordSet(records, dc, nil)
+	if err != nil {
+		return nil, err
+	}
 
-		differ := diff.New(dc)
-		_, create, del, modify, err := differ.IncrementalDiff(curRecords)
-		if err != nil {
-			return nil, err
+	for _, change := range instructions {
+		switch change.Type {
+		case diff2.DELETE:
+			corrections = append(corrections, change.CreateCorrection(wrapChangeFunction(change.Old, func(rec domain.DNSEntry) error { return n.domains.RemoveDNSEntry(dc.Name, rec) })))
+		case diff2.CREATE:
+			corrections = append(corrections, change.CreateCorrection(wrapChangeFunction(change.New, func(rec domain.DNSEntry) error { return n.domains.AddDNSEntry(dc.Name, rec) })))
+		case diff2.CHANGE:
+	        if canDirectApplyDNSEntries(change) {
```

Comment:
(this is my first attempt at using the "suggest edit" feature with the web-based VSCode. As you can see, it committed the change instead of doing it as a suggestion.  My apologies.  Feel free to revert the commit if you dislike this change.  Anyway...

I've moved the call to canDirectApplyDNSEntries() to the diff2.CHANGE case. This reduces the indent level and is more readable.

Reply:
No worries, I figured as much. I like the change, I mean, I don't like the conditional depth, but that didn't change between our commits , at least yours makes it realllly clear `canDirectApplyDNSEntries` applies to changes only.

For future references, in github and gitlab you can select multiple lines by dragging on the gutter , than if you click the "suggest" button. It will insert the code snippet you selected, if you make changes in that code snippet, it will post them as suggestion.

Reply:
Yes, I've used the "suggest edit" before.  However I was trying to do it with a different editor.  Did you know if you type `.` while looking at a PR, it starts up VS Code?

Reply:
yeah I did, I use it if I want to make quick edits / commits. Really nice stuff.